File: xssfinder-java-executor/src/main/java/org/xssfinder/runner/CustomNormalTraversalStrategy.java
package org.xssfinder.runner;

import org.xssfinder.CustomTraverser;
import org.xssfinder.remote.TraversalMode;

import java.lang.reflect.Method;
import java.util.HashMap;
import java.util.Map;

public class CustomNormalTraversalStrategy implements TraversalStrategy {
    private final CustomTraverserInstantiator traverserInstantiator;

    public CustomNormalTraversalStrategy(CustomTraverserInstantiator traverserInstantiator) {
        this.traverserInstantiator = traverserInstantiator;
    }

    @Override
    public boolean canSatisfyMethod(Method method, TraversalMode traversalMode) {
        if (traversalMode != TraversalMode.NORMAL) {
            return false;
        }
        CustomTraverser customTraverser =  traverserInstantiator.instantiate(method);
        return customTraverser != null;
    }

    @Override
    public TraversalResult traverse(Object page, Method method) {
        CustomTraverser customTraverser =  traverserInstantiator.instantiate(method);
        Map<String, String> inputIdsToAttackIds = new HashMap<String, String>();
        Object newPage = customTraverser.traverse(page);
        return new TraversalResult(newPage, inputIdsToAttackIds);
    }
}

File: xssfinder-php-executor/src/XssFinder/Runner/CustomNormalTraversalStrategy.php
<?php

namespace XssFinder\Runner;

use ReflectionMethod;
use XssFinder\TraversalMode;
use XssFinder\TUntraversableException;

class CustomNormalTraversalStrategy implements TraversalStrategy
{
    /** @var CustomTraverserInstantiator */
    private $_instantiator;

    function __construct(CustomTraverserInstantiator $instantiator)
    {
        $this->_instantiator = $instantiator;
    }

    /**
     * @param ReflectionMethod $method
     * @param int $traversalMode
     * @return boolean
     */
    function canSatisfyMethod(ReflectionMethod $method, $traversalMode)
    {
        return $traversalMode === TraversalMode::NORMAL && $this->_instantiator->hasCustomTraverser($method);
    }

    /**
     * @param mixed $page
     * @param ReflectionMethod $method
     * @return TraversalResult
     *
     * @throws TUntraversableException
     */
    function traverse($page, ReflectionMethod $method)
    {
        $customTraverser = $this->_instantiator->instantiate($method);
        $newPage = $customTraverser->traverse($page);
        return new TraversalResult($newPage, array());
    }
}